RFS/src/main/java/org/opensearch/migrations/bulkload/ManuallyManagedLuceneShardRepository.java

```diff
@@ -0,0 +1,128 @@
+package org.opensearch.migrations.bulkload;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.opensearch.migrations.bulkload.lucene.LuceneIndexReader;
+import org.opensearch.migrations.bulkload.lucene.version_9.IndexReader9;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ManuallyManagedLuceneShardRepository extends LuceneBasedDocumentRepository {
```

Comment:
I'm not sure what to make of this class, it seems like it belongs in its own top level directory as it seems to be an independent tool.  What do you think of adding an explanation / data flow diagram of how this will be used and in what case in conjunction with the Metadata/RFS workflow?